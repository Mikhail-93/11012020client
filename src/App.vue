<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <v-list dense>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-home</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Home</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-contact-mail</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Contact</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
                dark
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>Application</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height pa-0" fluid>
                <LMap ref="map" style="z-index: 0">
                    <l-tile-layer
                            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                            attribution='Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
                    >
                    </l-tile-layer>
                    <!--                    <l-control-attribution position="bottomright" prefix="leaflet OpenStreetMap Contributions" >-->

                </LMap>
                <!--                class="align-stretch"-->
            </v-container>
        </v-content>
    </v-app>
</template>

<script>
    import {LMap, LTileLayer, LMarker, LControlAttribution} from 'vue2-leaflet';

    export default {
        components: {
            LMap,
            LTileLayer,
            LMarker,
            LControlAttribution
        },
        props: {
            source: String,
        },
        data: () => ({
            drawer: null,
        }),
        mounted() {
            this.$nextTick(() => {
                // this.map=
                this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>