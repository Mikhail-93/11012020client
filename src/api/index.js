import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
