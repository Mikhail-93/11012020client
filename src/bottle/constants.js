/**
 * Created by alexey2baranov on 8/20/16.
 *
 * домены и редикт урлы
 * https://vk.com/apps?act=manage
 * дебаг на телефоне
 * https://developers.google.com/web/tools/chrome-devtools/remote-debugging?utm_source=dcc&utm_medium=redirect&utm_campaign=2016q3
 */
let constants = {
  "development": {
    sw: {
      delay: 15000,
    },
    messenger: {
      // идентификатор нашего приложения ВК
      // clientId: 7210289,
      // страница, на которую нужно перейти для аутентификации
      loginUrl: 'http://localhost:8081/connect/vkontakte',
      // с мобилки
      // loginUrl: 'http://192.168.43.84:8081/connect/vkontakte'
    },
    api: {
      path: "http://localhost:8081/api"
      // с мобилки
      // path: "http://192.168.43.84:8081/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  },
  "test": {
    sw: {
      delay: Number.MAX_SAFE_INTEGER,
    },
    messenger: {
      // clientId: 7210289,
      loginUrl: 'http://localhost:8082/connect/vkontakte',
      // redirectUrl: 'http://localhost:8082/login/check-vk'
    },
    api: {
      path: "http://localhost:8082/api"
    },
    di: {
      fetch: false,
      cookie: true
    }
  },
  "staging": {
    sw: {
      delay: 20000,
    },
    messenger: {
      loginUrl: 'https://staging.kopnik.org/connect/vkontakte',
      // clientId: 7210289,
      // redirectUrl: 'https://staging.kopnik.org/login/check-vk'
    },
    api: {
      path: "https://staging.kopnik.org/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  }
}

export default constants
