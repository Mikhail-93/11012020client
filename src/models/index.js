/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}