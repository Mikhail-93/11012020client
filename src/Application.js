import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
