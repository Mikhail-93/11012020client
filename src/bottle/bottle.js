import Bottle from "bottlejs"
import _ from "lodash"
// import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../application/Application"
import constants from './constants'
import api from "../api";
import CookieService from "./CookieService"
import MK from "../mk/mk";
import LocaleManager from "../locales/LocaleManager";
import ru from '../locales/ru'
import en from '../locales/en'

Bottle.config.strict = true
const bottle = new Bottle()

bottle.factory('logger', function loggerFactory() {
    //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
    // LoglevelPluginPrefix.reg(loglevel)
    // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
    loglevel.setLevel(loglevel.levels.DEBUG)
    if (process.env.NODE_ENV == "test") {
        loglevel.setLevel(loglevel.levels.DEBUG)
        // LoglevelPluginToString.apply(loglevel, {})
    }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
    return loglevel
})

bottle.constant('messages', {
    ru,
    en,
})

bottle.factory('env', function env() {
    return process.env.NODE_ENV || 'development'
})
bottle.service('localeManager', function localeManager() {
    const localeManager = new LocaleManager()
    localeManager.currentLocale = 'ru'//container.env === 'development' ? 'en' : 'ru'
    return localeManager
})
bottle.service('cookieService', CookieService, 'constants')
bottle.factory('api', function apiFactory(container) {
    if (!container.constants.di.fetch) {
        throw new Error('not supported')
    }
    return api
})
bottle.factory('VK', function vkFactory(container) {
    return process.env.NODE_ENV === 'test' ? MK : global.VK
})
bottle.factory('constants', function constantsFactory() {
    if (!process.env.NODE_ENV) {
        throw new Error("NODE_ENV is not defined")
    }

    let local = {}//require("./local.js")
    let result = _.merge({}, constants, local)[process.env.NODE_ENV]

    return result
})
if (!Application) {
    throw new Error("Application is undefined in bottle")
}
bottle.service('application', Application, 'logger')

/**
 * @callback fetch
 * @param {string} url
 * @param {Object?} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} api
 * @property {MK.class} VK
 * @property {CookieService} cookieService
 * @property {Object} constants
 * @property {Object} constants.api
 * @property {String} constants.api.path
 * @property {Object} constants.di
 * @property {Boolean} constants.di.fetch
 * @property {Number} constants.di.cookie
 * @property {Number} constants.messenger.clientId
 * @property {String} constants.messenger.loginUrl
 * @property {String} constants.messenger.redirectUrl
 * @property {String} constants.sw
 * @property {String} constants.sw.delay
 * @property {LocaleManager} localeManager
 * @property {string} env
 * @property {{ru, en}} messages
 */
const container = bottle.container
export {bottle, container}
