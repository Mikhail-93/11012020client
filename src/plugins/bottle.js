import Bottle from "bottlejs"
import _ from "lodash"
import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../Application"
import config from '../../config'
import fetchApi from "../bottle/fetchApi";
import fetchApiMock from "../bottle/fetchApi.mock";

Bottle.config.strict = true
const bottle = new Bottle()

/**
 * Настраивает контейнер
 * Для тестов необходим сброс и перенастройка чтобы соблюсти изолированность тестов
 *
 * @param {Object} options параметры по умолчанию соответствуют наиболее приемлемым для окружения значениям окружению
 * @param {Number=} options.cookie
 * @param {Boolean=} options.fetch
 */
Bottle.prototype.setup = function (options) {
    bottle.factory('defaultFetchApiOptions', function defaultFetchApiOptionsFactory(container) {
        const cookies = {
            1: 'PHPSESSID=user-1',
            2: 'PHPSESSID=eqlae6a1grpvcirpemfvebf806',
            3: 'PHPSESSID=user-3',
            4: 'PHPSESSID=user-4',
            5: 'PHPSESSID=user-5',
            6: 'PHPSESSID=user-6',
            7: 'PHPSESSID=user-7',
            8: 'PHPSESSID=user-8',
            9: 'PHPSESSID=user-9',
            10: 'PHPSESSID=user-10',
        }
        const result = {}
        if (container.config.di.cookie) {
            result.headers = {
                cookie: cookies[container.config.di.cookie]
            }
        }
        // console.log(result)
        return result
    })
    bottle.factory('fetchApi', function fetchApiFactory(container) {
        return container.config.di.fetch ? fetchApi : fetchApiMock
    })
    bottle.factory('config', function configFactory() {
        if (!process.env.NODE_ENV) {
            throw new Error("NODE_ENV is not defined");
        }

        let local = {}//require("./local.js")
        let result = _.merge({}, config, local)[process.env.NODE_ENV]

        return result
    })
    if (!Application){
        throw new Error("Application is undefined in bottle")
    }
    bottle.service('application', Application, 'logger')
    bottle.factory('logger', function loggerFactory() {
        //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
        // LoglevelPluginPrefix.reg(loglevel)
        // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
        loglevel.setLevel(loglevel.levels.DEBUG)
        if (process.env.NODE_ENV == "test") {
            loglevel.setLevel(loglevel.levels.WARN)
            // LoglevelPluginToString.apply(loglevel, {})
        }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
        return loglevel
    })
    Object.assign(this.container.config.di, options)
}

bottle.setup()

/**
 * @callback fetch
 * @param {string} url
 * @param {Object} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} fetchApi
 * @property {Object} defaultFetchApiOptions
 * @property {Object} defaultFetchApiOptions.headers
 * @property {Object} defaultFetchApiOptions.headers.cookie
 * @property {Object} config
 */
const container = bottle.container
export {bottle, container}
