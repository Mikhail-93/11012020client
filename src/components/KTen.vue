<template>
    <v-container fluid class="fill-height k-ten flex-column align-center">
        <v-card elevation="12" class="mb-10" width="100%" max-width="350px">
            <!--            <v-card-title>
                            Моя десятка
                        </v-card-title>-->
            <div class="d-flex flex-wrap mt-2 justify-space-around">
                <template v-for="(eachDruzhe, eachDruzheIndex) of extendedTen">
                    <v-badge v-if="eachDruzhe" :key="eachDruzhe.id" :title="eachDruzhe.name"
                             class='k-badge-event-handler' color="red" :offset-x="(64/64)*7+14" :offset-y="(64/64)*7+14"
                    >
                        <div @click="druzhe_del(eachDruzhe)" slot="badge" style="cursor: pointer">x</div>
                        <AvatarVue :value="eachDruzhe"
                                   :size="64" class="mb-2">
                        </AvatarVue>
                    </v-badge>
                    <v-avatar v-else :key="-eachDruzheIndex"

                 :size="64" class="mb-2">
                        <v-icon :size="64">mdi-account</v-icon>
                    </v-avatar>
                </template>
            </div>
        </v-card>

        <transition-group v-if="application.user.tenRequests && application.user.tenRequests.length" name="list-complete" tag="div" class="mx-auto"
                          style="width: 100%; max-width:350px; position: relative;">
            <v-card v-for="eachRequest in value.tenRequests" :key="eachRequest.id"
                    elevation="12" class="mb-10 list-complete-item" style="width: 100%;">
                <kopnik-view :value="eachRequest" location></kopnik-view>
                <v-divider></v-divider>
                <v-card-actions>
                    <v-btn color="success" class="flex"
                           @click="request_confirm(eachRequest)">
                        Принять в десятку
                    </v-btn>
                    <v-btn color="error" class="flex"
                           @click="request_decline(eachRequest)">
                        Отклонить
                    </v-btn>
                </v-card-actions>
            </v-card>
        </transition-group>
        <div v-else class="overline font-weight-bold my-auto" style="font-size: 1.625rem !important; color: #BDBD">Заявок нет</div>
    </v-container>
</template>
<script>
    import Kopnik from "../models/Kopnik"
    import thanks from "../thanks";

    import KopnikView from './KopnikVue'
    import AvatarVue from "./AvatarVue";
    import logger from "./mixin/logger";
    import {container} from "../plugins/bottle";

    export default {
        name: "Ten",
        mixins: [logger],
        components: {
            KopnikView,
            AvatarVue,
        },
        data: () => {
            return {
                application: container.application,
                items: [1, 2, 3, 4, 5, 6, 7, 8, 9],
            }
        },
        props: {
            value: {
                type: Kopnik,
                required: true
            }
        },
        computed: {
            extendedTen() {
                const result = [],
                    ten = this.application.user.ten
                for (let length = 0; length < 10; length++) {
                    result.push((ten !== undefined && ten.length >= length) ? ten[length] : null)
                }
                return result
            }
        },
        methods: {
            shuffle: function () {
                this.items.splice(0, 1)
                // this.items = _.shuffle(this.items)
            },
            request_confirm(request) {
                this.shuffle()
                const requests = this.application.user.tenRequests
                requests.splice(requests.indexOf(request), 1)
                this.application.user.ten.push(request)
            },
            request_decline(request) {
                const requests = this.application.user.tenRequests
                requests.splice(requests.indexOf(request), 1)
            },
            druzhe_del(druzhe) {
                const ten = this.application.user.ten
                ten.splice(ten.indexOf(druzhe), 1)
            }
        }
        ,
        async created() {
            // this.application.user.ten = [Kopnik.getReference(1), Kopnik.getReference(3), Kopnik.getReference(4)]
            this.application.user.tenRequests = [
                Kopnik.getReference(1), Kopnik.getReference(3), Kopnik.getReference(4),
            ]
        }
        ,
    }
</script>

<style>
    .k-badge-event-handler .v-badge__wrapper{
        pointer-events: all;
    }

    .list-complete-leave-active {
        position: absolute;
    }

    .list-complete-leave-to {
        opacity: 0;
        transform: translateX(100px);
    }

    .list-complete-item {
        transition: all .25s;
        display: inline-block;
        margin-right: 10px;
    }
</style>
