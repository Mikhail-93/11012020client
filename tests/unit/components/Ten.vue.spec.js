import flushPromises from "flush-promises";
import Vue from 'vue'

import vuePlugins from "../../test-setup";
import KTen from "../../../src/components/KTen";
import {bottle, container} from "../../../src/bottle/bottle";
import {Kopnik} from "../../../src/models";

describe('unit components Ten', () => {
    beforeAll(async () => {
        await login(6)
    })

    beforeEach(() => {
    })

    it.skip('draw', async () => {
        const kopnik= Kopnik.getReference(6)
        await container.application.authenticate()
        kopnik.ten= [7,].map(eachDruzheId=>Kopnik.getReference(eachDruzheId))
        kopnik.witnessRequests= [1,].map(eachDruzheId=>Kopnik.getReference(eachDruzheId))
        const vm = new Vue({
            ...KTen,
            ...vuePlugins,
            propsData:{
                value: kopnik
            }
        })
        vm.$mount()
        await flushPromises()
        expect(vm.$el).toMatchSnapshot()
    })
})

