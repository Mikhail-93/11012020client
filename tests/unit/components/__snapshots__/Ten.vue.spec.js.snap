// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit components Ten draw 1`] = `
<div
  class="container fill-height k-ten flex-column align-center container--fluid"
>
  <div
    class="mb-10 v-card v-sheet theme--light elevation-12"
    style="max-width: 350px; width: 100%;"
  >
    <div
      class="d-flex flex-wrap mt-2 justify-space-around"
    >
      <span
        class="v-badge k-badge-event-handler theme--light"
        title="Бобров Вячеслав Кудрявцевa"
      >
         
        <span
          class="v-badge mb-2 v-badge--bottom theme--light"
        >
          <div
            class="v-avatar"
            style="height: 64px; min-width: 64px; width: 64px;"
          >
            <img
              src="/avatar.png"
              style="object-fit: cover;"
            />
          </div>
          <span
            class="v-badge__wrapper"
          >
            <span
              aria-atomic="true"
              aria-label="знак"
              aria-live="polite"
              class="v-badge__badge orange"
              role="status"
            >
              1
            </span>
          </span>
        </span>
        <span
          class="v-badge__wrapper"
        >
          <span
            aria-atomic="true"
            aria-label="знак"
            aria-live="polite"
            class="v-badge__badge red"
            role="status"
            title="Бобров Вячеслав Кудрявцевa"
          >
            <div
              style="cursor: pointer;"
            >
              x
            </div>
          </span>
        </span>
      </span>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
      <div
        class="v-avatar mb-2"
        style="height: 64px; min-width: 64px; width: 64px;"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mdi mdi-account theme--light"
          style="font-size: 64px;"
        />
      </div>
    </div>
  </div>
   
  <div
    class="mx-auto"
    style="width: 100%; max-width: 350px; position: relative;"
  >
    <div
      class="mb-10 list-complete-item v-card v-sheet theme--light elevation-12"
      style="width: 100%;"
    >
      <!---->
       
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions"
      >
        <button
          class="flex v-btn v-btn--contained theme--light v-size--default success"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
                    Принять в десятку
                
          </span>
        </button>
         
        <button
          class="flex v-btn v-btn--contained theme--light v-size--default error"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
                    Отклонить
                
          </span>
        </button>
      </div>
    </div>
    <div
      class="mb-10 list-complete-item v-card v-sheet theme--light elevation-12"
      style="width: 100%;"
    >
      <!---->
       
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions"
      >
        <button
          class="flex v-btn v-btn--contained theme--light v-size--default success"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
                    Принять в десятку
                
          </span>
        </button>
         
        <button
          class="flex v-btn v-btn--contained theme--light v-size--default error"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
                    Отклонить
                
          </span>
        </button>
      </div>
    </div>
    <div
      class="mb-10 list-complete-item v-card v-sheet theme--light elevation-12"
      style="width: 100%;"
    >
      <!---->
       
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions"
      >
        <button
          class="flex v-btn v-btn--contained theme--light v-size--default success"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
                    Принять в десятку
                
          </span>
        </button>
         
        <button
          class="flex v-btn v-btn--contained theme--light v-size--default error"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
                    Отклонить
                
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
